src/engine/strat_engine/pool.rs

```diff
@@ -101,11 +106,22 @@ impl StratPool {
         let time = time::now().to_timespec();
 
         for (_, bd) in &mut self.block_devs {
-            bd.save_state(&time, data).unwrap(); // ignoring failure
+            bd.save_state(&time, data.as_bytes()).unwrap(); // ignoring failure
```

Comment:
Please change this comment to a TODO or something. As it stands it is deeply ambiguous. I'm assuming that you're against just panicking and crashing if saving the metadata fails on a block device, but it is actually not obvious from the comment.